src/main/java/org/spongepowered/common/mixin/core/block/BlockLiquidMixin.java

```diff
@@ -54,16 +55,18 @@
     )
     private void impl$CheckForLiquidMixing(final World worldIn, final BlockPos pos, final IBlockState state,
         final CallbackInfoReturnable<Boolean> cir, final boolean flag, final Integer integer) {
-        final IBlockState newState = integer == 0 ? Blocks.OBSIDIAN.getDefaultState() : Blocks.COBBLESTONE.getDefaultState();
-        final ChangeBlockEvent.Modify event = SpongeCommonEventFactory.callChangeBlockEventModifyLiquidMix(worldIn, pos, newState, null);
-        final Transaction<BlockSnapshot> transaction = event.getTransactions().get(0);
-        if (event.isCancelled() || !transaction.isValid()) {
-            cir.setReturnValue(false);
-            return;
-        }
-        final boolean success = worldIn.setBlockState(pos, (IBlockState) transaction.getFinal().getState());
-        if (!success) {
-            cir.setReturnValue(false);
+        if (ShouldFire.CHANGE_BLOCK_EVENT_MODIFY) {
```

Comment:
Instead of indenting this whole bit, check if it's false and return/short-circuit.